book/src/developers/designs/application_passwords.md

```diff
@@ -0,0 +1,229 @@
+where to go to use their app password?)
+
+(NOTE: To make it easier to manage tokens, it could be good to have a
+"token-capable" class type to prevent duplication this with service-accounts,
+and allows adding tokens to future types).
+
+### Accounts
+
+The user may wish to have multiple passwords per application. Each password must
+have a label to identify it, a enabled/disabled flag and a last used timestamp.
+For example:
+
+```
+  MAIL
+    iphone: abcd...
+    laptop: bcde...
+  HTTP
+    workstation: cdef...
+```
+
+Person accounts will need a new `Attribute::ApplicationPassword` that stores a
+`ValueSetApplicationPassword`. Each value in the set is a new type to manage these
+secrets and their labeling and the references to the applications.
+
+```
+struct ApplicationPAssword {
+    label: String,
+    enabled: bool,
+    last_used: Duration,
+    password: Password,
+}
+
+type ApplicationPasswords = BTreeMap<Uuid, ApplicationPassword>;
+                                     ^        ^
+                                     |        |
+                                     |        +-> Application password
+                                     +-> Application password UUID
+
+type ApplicationUuid = Uuid;
+
+struct ValueSetApplicationPassword {
+  map: Map<ApplicationUuid, ApplicationPasswords>
+}
+```
+
+Each value in the set is queried by its UUID. This defines the value as
+`Value::ApplicationPassword(Uuid, ApplicationPassword)`. The
+`ApplicationPassword` type implements `PartialEq` so two application passwords
+are equal if their label is equal and they refer to the same application.
+
+The user must be able to delete credentials individually. The generated password
+is only displayed once when the user creates it and it is not possible to
+recover the clear-text form, only hashed form is stored. It is not allowed to
+store duplicated application passwords (same app refer and label).
+
+Modifying the enabled/disabled flag will update the last_used field.
+
+We do not need temporary locks or holds - users can delete and recreate as
+needed.
+
+There will be a system-level cleanup task to delete disabled application
+passwords after a configurable time since last used.
+
+### Reference integrity
+
+Since application passwords are related to applications, on delete of an
+application all entries that have a bound application password should be removed
+from user accounts.
+
+### Access controls
+
+The "Application administrators" group will manage the applications, and
+applications will allow "managed by" so that they can have delegated
+administration.
+
+The "Application user passwords administrators" group will be able to list the
+users's application passwords and delete them but only the users will be able
+to, additionally to listing and deleting, self-create their own application
+passwords.
+
+### LDAP
+
+The rootdse needs to be extended to include the applications as additional
```

Comment:
I don't think so, because this could get very noisy very fast, and clients could then try to walk all the different DIT's

Reply:
OK, removed.